packages/openid4vp/src/openid4vp-auth-request/create-openid4vp-auth-request.ts

```diff
@@ -0,0 +1,71 @@
+import type { CallbackContext, JwtSigner } from '@openid4vc/oauth2'
+import { uriEncodeObject } from '@openid4vc/utils'
+import { createJarAuthRequest } from '../jar/create-jar-auth-request'
+import type { Openid4vpAuthRequest } from './v-openid4vp-auth-request'
+import { validateOpenid4vpAuthRequestParams } from './validate-openid4vp-auth-request'
+
+/**
+ * Creates an OpenID4VP authorization request, optionally with a JWT Secured Authorization Request (JAR)
+ * If the request is created after receiving wallet metadata via a POST to the request_uri endpoint, the wallet nonce needs to be provided
+ *
+ * @param input Configuration options for creating the authorization request
+ * @param input.scheme Optional URI scheme to use (defaults to 'openid4vp://')
+ * @param input.requestParams The OpenID4VP authorization request parameters
+ * @param input.jar Optional JWT Secured Authorization Request (JAR) configuration
+ * @param input.jar.requestUri The URI where the JAR will be accessible
+ * @param input.jar.jwtSigner Function to sign the JAR JWT
+ * @param input.jar.jweEncryptor Optional function to encrypt the JAR JWT
+ * @param input.jar.additionalJwtPayload Optional additional claims to include in JAR JWT
+ * @param input.wallet Optional wallet-specific parameters
+ * @param input.wallet.nonce Optional wallet nonce
+ * @param input.callbacks Callback functions for JWT operations
+ * @returns Object containing the authorization request parameters, URI and optional JAR details
+ */
+export async function createOpenid4vpAuthorizationRequest(input: {
+  scheme?: string
+  requestParams: Openid4vpAuthRequest
+  jar?: {
+    requestUri: string
+    jwtSigner: JwtSigner
+    jweEncryptor?: JwtSigner
+    additionalJwtPayload?: Record<string, unknown>
+  }
+  wallet?: {
+    nonce?: string
+  }
+  callbacks: Pick<CallbackContext, 'signJwt' | 'encryptJwe'>
```

Comment:
With this many input parameters I generally like to define an interface, as you can then also add the comments nicely inline in the interface and it will also show when hovering:

<img width="440" alt="image" src="https://github.com/user-attachments/assets/0f81611d-9126-483a-8929-395dc9b13742" />